### ISSUE TITLE
Weird `required configuration keys were missing` error

### ISSUE THREAD

**Issue by doitadrian** [pulumi/pulumi-aws#1153]
Hi all,

not sure if this is a bug but...

When I don't have `config.aws:xyz` in my `Pulumi.{STACK}.yaml`, I get the following error:
![image](https://user-images.githubusercontent.com/5121148/95352105-2e654b00-08c2-11eb-9017-48c2e987d141.png)

So basically, I'm forced to set something into it, just to avoid the error.

BTW, the versions are (my `package.json`):
![image](https://user-images.githubusercontent.com/5121148/95352441-8a2fd400-08c2-11eb-826a-3a1aece58e60.png)

And the CLI version is `2.11.2`.

**Comment by stack72** [comment]
Hi @doitadrian 

The issue is because you have not set `aws:region`

I have an issue opened in the underlying bridge to fix this - https://github.com/pulumi/pulumi-terraform-bridge/issues/282

I hope this makes sense

Paul

**Comment by doitadrian** [comment]
Thanks for a really quick response @stack72. 🤯

Yeah, that's true, I don't have `aws:region`, and there is a reason for that, I actually pass it as the `AWS_REGION` env variable.

Hope that helps clarify things a bit.

**Comment by stack72** [comment]
So Pulumi actually requires the region to be set as a configuration property - this is due to a legacy design decision. It is the only property that was required to be set - the rest can be read from environment variables

**Comment by doitadrian** [comment]
Note that I was actually deploying resources with the region set via the `AWS_REGION` env variable for some time now, no problems were caused by doing that. But note - I was doing it with the `2.x.y` version of the `@pulumi/aws` package.

In other words, the only reason I'm reporting this is that this error didn't pop up in `2.x.y` version, which I was using before. I started seeing it once I updated to `3.5.0`, which I did just now.

And interestingly enough, the problem goes away by setting an arbitrary value into an arbitrary `aws:xyz` configuration property, and everything still works (proper regions are being used upon deploying resources).

I will just set an arbitrary value for now and monitor it for some time, just to make sure everything works as expected.
